UnityProject/Assets/Scripts/Player/Huggable.cs

```diff
@@ -82,8 +96,46 @@ private bool TryFieryHug()
 		return false;
 	}
 
+	private void Judgement(LivingHealthMasterBase puller)
+	{
+		if (lastPuller == null || lastPuller != puller)
+		{
+			JudgementCooldown(puller);
+			return;
+		}
+		if (DMMath.Prob(tailPullJudgementChance))
+		{
+			puller.Gib();
+		}
+	}
+
+	private async void JudgementCooldown(LivingHealthMasterBase puller)
+	{
+		lastPuller = puller;
+		await Task.Delay(tailPullJudgementCooldown).ConfigureAwait(false);
+		lastPuller = null;
+	}
```

Comment:
We have a beautiful cooldowns helper class just for this purpose : )

Something like `Cooldowns.TryStart(...)` turns all this into a one-liner.